src/base.jl

```diff
@@ -7,11 +7,25 @@
 # -
 
 
+"""
+    AbstractIVP{T,Y}
+
+A progenitor of types representing an IVP (initial value
+problem).  The type parameters `T` and `Y` correspond to the types of
+time and state variable respectively.
+
+"""
 abstract AbstractIVP{T,Y}
-Base.eltype{T,Y}(::Type{AbstractIVP{T,Y}}) = T,Y,Y
+
+"""
+The "elements" of AbstractIVP are `t,y,dy` so the `eltype` returns `Tuple{T,Y,Y}`
+"""
+Base.eltype{T,Y}(::Type{AbstractIVP{T,Y}}) = Tuple{T,Y,Y}
 
 """
 
+    IVP{T,Y,F,G,J} <: AbstractIVP{T,Y}
```

Comment:
I looked it up and tested it.  It seems it's not working properly in all cases and the produced output is not very clear (e.g. we would like to highlight some optional arguments and their defaults).  I would leave it as is for now.